### PR TITLE
Pass pod hostname as system-uid option to run_single_worker.rb

### PR DESCRIPTION
**Pull request by jrafanie**
- [x] Depends on [new system_uid column in schema](https://github.com/ManageIQ/manageiq-schema/pull/533)

Needed for https://github.com/ManageIQ/manageiq/pull/20792

### CODE REVIEW

File: images/manageiq-base-worker/container-assets/entrypoint [hunk 1]
```diff
@@ -11,6 +11,8 @@ KEY
 
 echo "${GUID}" > ${APP_ROOT}/GUID
 
-[[ -n $EMS_IDS ]] && WORKER_OPTIONS="--ems-id=${EMS_IDS} "
+WORKER_OPTIONS=""
+[[ -n $EMS_IDS ]] && WORKER_OPTIONS="${WORKER_OPTIONS} --ems-id=${EMS_IDS} "
+[[ -n $HOSTNAME ]] && WORKER_OPTIONS="${WORKER_OPTIONS} --system-uid=$HOSTNAME "
```
Comment (Fryguy) [inline images/manageiq-base-worker/container-assets/entrypoint:16]: This might affect docker monolithic, so we should just verify.

**Comment by jrafanie** [comment]
I've been reviewing travis this afternoon and it looks like the ppc builds are failing on travis.com but not travis.org [my fork's branch](https://travis-ci.org/github/jrafanie/manageiq-pods/jobs/742555236) and it's also failing on kasparov on the main repo.  I'd imagine master will also fail when something gets merged.

File: images/manageiq-base-worker/container-assets/entrypoint (hunk 1 above)
Comment (jrafanie) [reply images/manageiq-base-worker/container-assets/entrypoint:16]: Do we still support monolithic on master/k?  I have no idea how to build or test it.

Comment (Fryguy) [reply images/manageiq-base-worker/container-assets/entrypoint:16]: Yes, well, we "support" it 😄 .  It's more meant for "trying" out ManageIQ.

Comment (Fryguy) [reply images/manageiq-base-worker/container-assets/entrypoint:16]: if you can build the pods images, then it's just a regular Docker build from the core repo - it depends on the manageiq-ui-worker image - https://github.com/ManageIQ/manageiq/blob/master/Dockerfile#L2

**Comment by jrafanie** [comment]
> I've been reviewing travis this afternoon and it looks like the ppc builds are failing on travis.com but not travis.org [my fork's branch](https://travis-ci.org/github/jrafanie/manageiq-pods/jobs/742555236) and it's also failing on kasparov on the main repo. I'd imagine master will also fail when something gets merged.

It's terribly slow on ppc64le builds partly because ruby tar.bz2 are coming from cdn on amd64 and from travis for ppc64le:

```
on amd64, it's instant
Found remote file https://rvm_io.global.ssl.fastly.net/binaries/ubuntu/16.04/x86_64/ruby-2.6.6.tar.bz2
Checking requirements for ubuntu.
Requirements installation successful.
ruby-2.6.6 - #configure
ruby-2.6.6 - #download
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 18.1M  100 18.1M    0     0  17.5M      0  0:00:01  0:00:01 --:--:-- 17.5M
```

```

Found remote file https://rubies.travis-ci.org/ubuntu/16.04/ppc64le/ruby-2.6.6.tar.bz2
```

^^ This takes > 200 seconds on ppc64le builds.

The second commit adds a `travis_wait` on the curl of the operator-sdk download and allows the build to complete.  It's a hack but gets the build to pass.

File: images/manageiq-base-worker/container-assets/entrypoint (hunk 1 above)
Comment (jrafanie) [reply images/manageiq-base-worker/container-assets/entrypoint:16]: I'll try to build it tomorrow.

Comment (kbrock) [reply images/manageiq-base-worker/container-assets/entrypoint:16]: is it viable to drop support for the monolith?
would be nice if we could reduce the number of different packaging options that we have.

Comment (Fryguy) [reply images/manageiq-base-worker/container-assets/entrypoint:16]: No, the monolith is used by a lot of people, including myself for trying things out...it's the simplest route as it's just `docker run manageiq/manageiq`

Comment (Fryguy) [reply images/manageiq-base-worker/container-assets/entrypoint:16]: Worked with this with LJ, and the key insight is that the system_uid value is being set in the _entrypoint_, however the entrypoint is overwritten with a custom on in the docker monolithic image.

Everything looks good :)

**Comment by simaishi** [comment]
Kasparov backport details:

```
$ git log -1
commit 00efede955f3d0e3f25c7f71acb0a5b146d4ad6c
Author: Jason Frey <fryguy9@gmail.com>
Date:   Wed Nov 11 17:07:28 2020 -0500

    Merge pull request #651 from jrafanie/set_system_uid_in_worker_pods

    Pass pod hostname as system-uid option to run_single_worker.rb

    (cherry picked from commit c0fa7dd94f73b5a0f936d30258ab09e2f41ac051)
```

**Comment by simaishi** [comment]
Jansa backport details:

```
$ git log -1
commit e696a392110c64115e5d4f05fdacc4bedf8b272b
Author: Jason Frey <fryguy9@gmail.com>
Date:   Wed Nov 11 17:07:28 2020 -0500

    Merge pull request #651 from jrafanie/set_system_uid_in_worker_pods

    Pass pod hostname as system-uid option to run_single_worker.rb

    (cherry picked from commit c0fa7dd94f73b5a0f936d30258ab09e2f41ac051)
```

**Pull request merged**